pkg/controller/deployment/deployment_controller.go

```diff
@@ -340,6 +396,10 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	if dc.garbageCollectorEnabled {
```

Comment:
@kargakis do you mean this RLock: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/thread_safe_store.go#L99-L101